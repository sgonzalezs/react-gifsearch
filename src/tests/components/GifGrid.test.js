import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
// Simulamos una peticion del custom hook
jest.mock('../../hooks/useFetchGifs');

describe('Tests file GifGrid', () => {

    const category='software';

    test('show component', () => {
        //Simulamos el objeto que retorna el custom hook
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper=shallow( <GifGrid category={category} /> );
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('Must show items when images are loaded', () => {
        
        const images=[{
            id:'ABC',
            url:'http://localhost/sample.jpg',
            title:'Sample title'
        }];
        useFetchGifs.mockReturnValue({
            data:images,
            loading:false
        });
        const wrapper=shallow( <GifGrid category={category} /> );
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(images.length);
    });
    
});
