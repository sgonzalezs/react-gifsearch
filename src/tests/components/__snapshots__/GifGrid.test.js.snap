// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests file GifGrid Must show items when images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    software
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Sample title"
      url="http://localhost/sample.jpg"
    />
  </div>
</Fragment>
`;

exports[`Tests file GifGrid show component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    software
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
